### ISSUE TITLE
Generated InputVector (IV) is a 0 length bytearray and it takes ages to figure out why

### ISSUE THREAD

**Issue by PJanissen** [jasypt/jasypt#8]
When trying to generate an encrypted password using the encrypt.bat cli, I got the following 'error':

> Operation not possible (Bad input or parameters)

This when running the following command:

`~\Downloads\jasypt-1.9.3-dist\jasypt-1.9.3\bin\encrypt.bat input="encryptme" password=abcd1234 algorithm=PBEWITHHMACSHA224ANDAES_256`

My initial thought was that this was due to my jvm not supporting either SHA or AES. So I messed around with it, making no difference. So ultimately, I cloned the repo and debugged the project.

Finally, I discovered that there was an internal error hidden away very deeply (StandardPBEByteEncryptor.class):

> java.security.InvalidAlgorithmParameterException: Wrong IV length: must be 16 bytes long

This happens because an iv is generated by the ivGenerator, which by default is a NoIvGenerator. Which returns a byte[] with length 0. Regardless of the length it is given.

To fix this, I have to pass along an extra command line argument 
> ivGeneratorClassName=org.jasypt.iv.RandomIvGenerator

So now, after 5 hours of debugging, I can generate my encrypted passwords.

My thoughts:

1. Why is this not covered in the documentation for the encryption CLI util. Only that the argument is there, not that it is a requirement for some algorithms.
2. Why pass a classname? Wouldn't it be more convenient to use a predefined set of choices?
3. Help a developer with some sort of feedback. I mean, I can understand this:

> // If encryption fails, it is more secure not to return any 
> // information about the cause in nested exceptions. Simply fail. 

But it is an absolute disaster to figure out what is going wrong.

Please make changes so other developers don't get stuck on this for as long as I have been.

**Comment by zx87876** [comment]
six six  six

**Comment by merusso** [comment]
Yep, I got stuck on this also. If not for stumbling upon this issue via a [StackOverflow answer](https://stackoverflow.com/a/61228950/428628), I would've given up on the client.

**Comment by lonely-one** [comment]
> When trying to generate an encrypted password using the encrypt.bat cli, I got the following 'error':
> 
> > Operation not possible (Bad input or parameters)
> 
> This when running the following command:
> 
> `~\Downloads\jasypt-1.9.3-dist\jasypt-1.9.3\bin\encrypt.bat input="encryptme" password=abcd1234 algorithm=PBEWITHHMACSHA224ANDAES_256`
> 
> My initial thought was that this was due to my jvm not supporting either SHA or AES. So I messed around with it, making no difference. So ultimately, I cloned the repo and debugged the project.
> 
> Finally, I discovered that there was an internal error hidden away very deeply (StandardPBEByteEncryptor.class):
> 
> > java.security.InvalidAlgorithmParameterException: Wrong IV length: must be 16 bytes long
> 
> This happens because an iv is generated by the ivGenerator, which by default is a NoIvGenerator. Which returns a byte[] with length 0. Regardless of the length it is given.
> 
> To fix this, I have to pass along an extra command line argument
> 
> > ivGeneratorClassName=org.jasypt.iv.RandomIvGenerator
> 
> So now, after 5 hours of debugging, I can generate my encrypted passwords.
> 
> My thoughts:
> 
> 1. Why is this not covered in the documentation for the encryption CLI util. Only that the argument is there, not that it is a requirement for some algorithms.
> 2. Why pass a classname? Wouldn't it be more convenient to use a predefined set of choices?
> 3. Help a developer with some sort of feedback. I mean, I can understand this:
> 
> > // If encryption fails, it is more secure not to return any
> > // information about the cause in nested exceptions. Simply fail.
> 
> But it is an absolute disaster to figure out what is going wrong.
> 
> Please make changes so other developers don't get stuck on this for as long as I have been.

Thank you for solving my long problem
